plugins/topology/src/hooks/useK8sObjectsResponse.test.ts

```diff
@@ -0,0 +1,45 @@
+import { useKubernetesObjects } from '@backstage/plugin-kubernetes';
+import { renderHook } from '@testing-library/react-hooks';
+import kubernetesObject from '../__fixtures__/kubernetesObject.json';
+import { watchResourcesData } from '../__fixtures__/k8sResourcesContextData';
+import { ModelsPlural } from '../models';
+import { useK8sObjectsResponse } from './useK8sObjectsResponse';
+
+const watchedResources = [
+  ModelsPlural.deployments,
+  ModelsPlural.pods,
+  ModelsPlural.services,
+  ModelsPlural.replicasets,
+];
+
+jest.mock('@backstage/plugin-kubernetes', () => ({
+  useKubernetesObjects: jest.fn(),
+}));
+
+const mockUseKubernetesObjects = useKubernetesObjects as jest.Mock;
+
+jest.mock('@backstage/plugin-catalog-react', () => ({
+  useEntity: () => ({
+    entity: {
+      metadata: {
+        name: 'test',
+      },
+    },
+  }),
+}));
+
+describe('useK8sObjectsResponse', () => {
```

Comment:
shall we add more tests here with `setSelectedCluster` and then check on the cluster if updated and one for `selectedClusterErrors` as well?